### PR TITLE
Add signing configurations for debug & release build types

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR addresses CI requirements to building a valid Android APK.

#### Changelist:
- Add signing configurations for debug & release build types
- Enabled code shrinking, obfuscation, and optimization for release mode
  - Re-added Proguard rules to ignore warnings related to these optimizations.

#### Other Changelist:
- Added additional comments
- Removed obsolete Proguard rules
  - These were tied to Firebase & JNA libraries used by Breez SDK notification service extension.
- Use " instead of ' char on `build.gradle`

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -46,11 +45,56 @@ android {
...
+    }
+
+    buildTypes {
+        debug {
             signingConfig = signingConfigs.debug
+            applicationIdSuffix = ".debug"
+            versionNameSuffix = "-DEBUG"
+        }
+        release {
+            signingConfig = signingConfigs.release
```
Comment (dangeross) [inline android/app/build.gradle:97]: This proguard file is for sure not needed here?

Comment (erdemyerebasmaz) [reply android/app/build.gradle:105]: I've re-added it for the default proguard rules to be used on release builds.

Comment (erdemyerebasmaz) [reply android/app/build.gradle:105]: @dangeross [Create proguard-rules.pro](https://github.com/breez/l-breez/pull/33/commits/adbc7274914eabfcad12fb35f61f5adeebcad103)

**Comment by erdemyerebasmaz** [comment]
> Looks good. Is it testable by running the build CI on the branch?

@dangeross Yes, I've triggered a [build](https://github.com/breez/l-breez/actions/runs/9713103762) to build an APK. We can merge after testing the created artifact.

**Comment by erdemyerebasmaz** [comment]
Tests have passed with CI build artifact. Merging.

**Pull request merged**